src/main/java/com/raisetech/mybatis/service/RamenServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.raisetech.mybatis.service;
+
+import com.raisetech.mybatis.controller.RamenResponse;
+import com.raisetech.mybatis.entity.Ramen;
+import com.raisetech.mybatis.mapper.RamenMapper;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class RamenServiceImpl implements RamenService {
+
+  private RamenMapper ramenMapper;
+
+  public RamenServiceImpl(RamenMapper movieMapper) {
+    this.ramenMapper = movieMapper;
+  }
+
+  public List<Ramen> findAll() {
+    return ramenMapper.findAll();
+  }
+
+  public List<Ramen> findByPoint(Integer point) {
+    return ramenMapper.getPoint(point);
+  }
+
+  public List<RamenResponse> getRamenList(Integer point) {
+    if (point != null) {
+      return findByPoint(point).stream().map(RamenResponse::new).toList();
+    } else {
+      return findAll().stream().map(RamenResponse::new).toList();
+    }
+  }
```

Comment:
2点ほどいい忘れていました。
RamenResponseですが、Serviceはあくまでも業務処理を扱うのでHTTPレスポンスに関するXxxResponseは使わないほうが違和感が少ないです。
XxxResponse系はController内でのみ利用するようにするのが私は良いと思います。

それからこれもコーディングルール次第ですが、Service内の1メソッドをアプリケーションのユースケースとして捉えてService内のメソッドが互いに依存しないようにするのがいいかなと思いました。

それらを踏まえると私ならこのように実装します。
```suggestion
  public List<Ramen> getRamenList(Integer point) {
    if (point != null) {
      return ramenMapper.getPoint(point);
    } else {
      return ramenMapper.findAll();
    }
  }
```

Reply:
どこに何を書くかという点は現場に入らないとなかなかわからないですし、現場によってはそこまで気にして実装しないということもあります 😓 

このコメントについて対応するかどうかはおまかせします。